Admin/AbstractAdmin.php

```diff
@@ -1531,6 +1532,23 @@ public function setPersistFilters($persist)
         );
 
         $this->persistFilters = $persist;
+
+        if ($persist === true) {
+            try {
+                $request = $this->getRequest();
+                $session = $request->getSession();
+
+                if ($session === null) {
+                    throw new \RuntimeException('The Request object has not session available');
+                }
+
+                $this->filterPersister = new SessionFilterPersister($session);
+            } catch (\RuntimeException $exception) {
+                // no request nor session
```

Comment:
So you accept that an exception will be thrown if this method is called when there is no request nor session ? (ie dependency injection for instance)